### ISSUE TITLE
Remove unneeded `_probabilities`?

### ISSUE THREAD

**Issue by emizzle** [codex-storage/codex-contracts-eth#137]
When a slot is filled, the proof probability is stored in a private mapping `_probabilities`. However, this is likely not needed as the proof probability for a `Request` is already stored in `Request.Ask`.

This may require passing in the request Id.

**Comment by AuHau** [comment]
Yeah, seems like it is not needed.

**Comment by markspanbroek** [comment]
Good idea. The mapping is currently used to to decouple Proofs.sol from needing to know about requests. But we might as well use a virtual function `slotProbability(SlotId)` for that, analogous to how we handle slot state: https://github.com/codex-storage/codex-contracts-eth/blob/fe8da1013d3b1270b3063147d8eb51550bd50f9c/contracts/Proofs.sol#L36
